source/distributed/tria_base.cc

```diff
@@ -163,6 +166,32 @@ namespace parallel
         return;
       }
 
+
+    {
+      // find ghost owners
+      for (typename Triangulation<dim,spacedim>::active_cell_iterator
+           cell = this->begin_active();
+           cell != this->end();
+           ++cell)
+        if (cell->is_ghost())
+          number_cache.ghost_owners.insert(cell->subdomain_id());
+
+      Assert(number_cache.ghost_owners.size() < Utilities::MPI::n_mpi_processes(mpi_communicator), ExcInternalError());
+    }
+
+    {
+      // find level ghost owners
+      for (typename Triangulation<dim,spacedim>::cell_iterator
+           cell = this->begin();
+           cell != this->end();
+           ++cell)
+        if (cell->level_subdomain_id() != numbers::artificial_subdomain_id
+            && cell->level_subdomain_id() != locally_owned_subdomain())
```

Comment:
1. There is a difference between ```ghost()``` and ```level_ghost()```: ```ghost()``` uses ```subdomain_id```, only works on active cells, and considers neighbors that are on different levels. ```level_ghost()``` uses the ```level_subdomain_id```, is defined on each level, and neighbors only exist on the same level.
2. There is currently no function ```level_ghost()``` so I have to write it out by hand.